### DISCUSSION TITLE
Servas with Portainer

### DISCUSSION THREAD

**Discussion by doc-huebi** [beromir/Servas#98] [Q&A]
Is there a way to get an APP_KEY when i'm using portainer to handle all my stacks? In that case, the

docker exec -it servas php artisan key:generate --force

is giving a dumo saying no .env file with APP_KEY, what is true of course.

Any other way to create an APP_KEY and put it nto the yaml?

**Comment by doc-huebi** [comment]
Installing it manually outside portainer, everything works fine, APP_KEY is now presentm but the only thing i'm geeting is a server 500 address :(

**Comment by beromir** [comment]
What do the logs say? You can check them with `docker logs servas`.

**Reply by neilkpd** [reply]
Starting Migration...

   Illuminate\Database\QueryException 

  SQLSTATE[HY000] [2002] Connection refused (Connection: mysql, SQL: select table_name as `name`, (data_length + index_length) as `size`, table_comment as `comment`, engine as `engine`, table_collation as `collation` from information_schema.tables where table_schema = 'forge' and table_type in ('BASE TABLE', 'SYSTEM VERSIONED') order by table_name)

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:829
    825▕                     $this->getName(), $query, $this->prepareBindings($bindings), $e
    826▕                 );
    827▕             }
    828▕ 
  ➜ 829▕             throw new QueryException(
    830▕                 $this->getName(), $query, $this->prepareBindings($bindings), $e
    831▕             );
    832▕         }
    833▕     }

      +39 vendor frames 

  40  artisan:35
      Illuminate\Foundation\Console\Kernel::handle()

Creating caches...

   INFO  Configuration cached successfully.  


   INFO  Blade templates cached successfully.  

[02-Feb-2025 20:52:42] NOTICE: fpm is running, pid 19
[02-Feb-2025 20:52:42] NOTICE: ready to handle connections
127.0.0.1 -  02/Feb/2025:20:52:54 +0000 "GET /index.php" 500

**Reply by beromir** [reply]
The application seems to have no connection to the MySQL container. Unfortunately I have no experience with Portainer. Do you have other services running with a separate database container? Are there any config files for the Docker containers or how do you manage them?

**Reply by beromir** [reply]
You can also use a SQLite database in the application container, without the need for an additional container.
https://github.com/beromir/Servas/tree/main/docker/sqlite-example

**Comment by neilkpd** [comment]
I have tried installing Servas with both databases, MySQL and  and I keep
getting - 'Connection refused'.
My setup:
TrueNAS Scale ElectricEel-24.10.2
Dockge
Servas is perfect for me and I would hate to give up!

Log:
servas  | Starting Migration...
servas  |
servas  |    Illuminate\Database\QueryException
servas  |
servas  |   SQLSTATE[HY000] [2002] Connection refused (Connection: mysql,
SQL: select table_name as `name`, (data_length + index_length) as `size`,
table_comment as `comment`, engine as `engine`, table_collation as
`collation` from information_schema.tables where table_schema = 'forge' and
table_type in ('BASE TABLE', 'SYSTEM VERSIONED') order by table_name)
servas  |
servas  |   at
vendor/laravel/framework/src/Illuminate/Database/Connection.php:829
servas  |     825▕                     $this->getName(), $query,
$this->prepareBindings($bindings), $e
servas  |     826▕                 );
servas  |     827▕             }
servas  |     828▕
servas  |   ➜ 829▕             throw new QueryException(
servas  |     830▕                 $this->getName(), $query,
$this->prepareBindings($bindings), $e
servas  |     831▕             );
servas  |     832▕         }
servas  |     833▕     }
servas  |
servas  |       +39 vendor frames
servas  |
servas  |   40  artisan:35
servas  |       Illuminate\Foundation\Console\Kernel::handle()
servas  |
servas  | Creating caches...
servas  |
servas  |    INFO  Configuration cached successfully.
servas  |
servas  |
servas  |    INFO  Blade templates cached successfully.
servas  |
servas  | [05-Feb-2025 18:08:49] NOTICE: fpm is running, pid 19
servas  | [05-Feb-2025 18:08:49] NOTICE: ready to handle connections

On Wed, Feb 5, 2025 at 2:50 AM Henrik ***@***.***> wrote:

> You can also use a SQLite database in the application container, without
> the need for an additional container.
> https://github.com/beromir/Servas/tree/main/docker/sqlite-example
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/beromir/Servas/discussions/98#discussioncomment-12064393>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AARYC2S7KEPFW6NWSTRLVHL2OG7CTAVCNFSM6AAAAABRKXU366VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTEMBWGQZTSMY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Regards,
Neil

**Reply by beromir** [reply]
Have you enabled SQLite in the .env file?
```
# SQLite
DB_CONNECTION=sqlite
DB_DATABASE=/var/www/html/database/sqlite/servas.db
DB_FOREIGN_KEYS=true
```

**Comment by neilkpd** [comment]
Hey.
These are the files with logs.
Hope you can see what I'm doing wrong!

Both of these files are in the same folder.

# docker-compose.yaml
services:
  servas:
    image: beromir/servas
    container_name: servas
    restart: unless-stopped
    ports:
      - "8080:80"
    volumes:
      - ./.env:/var/www/html/.env
      - /mnt/tank0/docker/servas/data:/var/www/html/database/sqlite

# .env
APP_NAME=Servas
APP_ENV=production
APP_KEY=
APP_DEBUG=false
APP_URL=http://192.168.1.30:8080

SERVAS_ENABLE_REGISTRATION=true

# SQLite
DB_CONNECTION=sqlite
DB_DATABASE=/var/www/html/database/sqlite/servas.db
DB_FOREIGN_KEYS=true

# Log
Starting Migration...
   INFO  Preparing database.
  Creating migration table ......................................... 11ms
DONE
   INFO  Loading stored database schemas.
  database/schema/sqlite-schema.dump .............................. 178ms
DONE
   INFO  Running migrations.
  2022_08_11_143315_add_two_factor_confirmed_column_to_users_table .. 6ms
DONE
  2022_11_26_194415_remove_media_type_column_from_links_table ...... 56ms
DONE
  2023_03_19_203332_add_expires_at_column_to_personal_access_tokens_table
 5ms DONE
  2023_04_30_173655_drop_later_table ................................ 5ms
DONE
  2024_02_19_210136_increase_link_column_size ...................... 19ms
DONE
  2024_03_02_180621_create_public_links_tables ..................... 21ms
DONE
  2024_04_05_185606_add_query_options_columns_to_groups_table ...... 11ms
DONE
Creating caches...
   INFO  Configuration cached successfully.
   INFO  Blade templates cached successfully.
[07-Feb-2025 12:00:05] NOTICE: fpm is running, pid 20
[07-Feb-2025 12:00:05] NOTICE: ready to handle connections
127.0.0.1 -  07/Feb/2025:12:01:23 +0000 "GET /index.php" 500
127.0.0.1 -  07/Feb/2025:12:01:30 +0000 "GET /index.php" 500
127.0.0.1 -  07/Feb/2025:12:01:54 +0000 "GET /index.php" 500
127.0.0.1 -  07/Feb/2025:12:02:19 +0000 "GET /index.php" 500

On Fri, Feb 7, 2025 at 4:14 AM Henrik ***@***.***> wrote:

> Have you enabled SQLite in the .env file?
>
> # SQLite
> DB_CONNECTION=sqlite
> DB_DATABASE=/var/www/html/database/sqlite/servas.db
> DB_FOREIGN_KEYS=true
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/beromir/Servas/discussions/98#discussioncomment-12091785>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AARYC2Q6QUNFGLWVVJRSIC32OR2PXAVCNFSM6AAAAABRKXU366VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTEMBZGE3TQNI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Regards,
Neil

**Reply by beromir** [reply]
Hey,
this looks good, the database connection is now working and the migrations are complete.
Now you have to generate an application key:
```bash
docker exec -it servas php artisan key:generate --force
```

**Comment by neilkpd** [comment]
The sqlite version is now working. Thank you.

I would really like to get the mariadb version working.
Here are my files with errors:

#docker-compose.yaml
services:
  db:
    image: mariadb:10.7.3
    container_name: servas_db
    restart: unless-stopped
    command: mysqld --character-set-server=utf8mb4
--collation-server=utf8mb4_bin
    environment:
      - MARIADB_ROOT_PASSWORD=${DB_PASSWORD}
      - MARIADB_USER=${DB_USERNAME}
      - MARIADB_PASSWORD=${DB_PASSWORD}
      - MARIADB_DATABASE=${DB_DATABASE}
    volumes:
      - /mnt/tank0/docker/servas/data:/var/lib/mysql

  servas:
    image: beromir/servas
    container_name: servas
    restart: unless-stopped
    depends_on:
      - db
    ports:
      - "8080:80"
    volumes:
      - ./.env:/var/www/html/.env

#.env
APP_NAME=Servas
APP_ENV=production
APP_KEY=
APP_DEBUG=false
APP_URL=http://192.168.1.30:8080

SERVAS_ENABLE_REGISTRATION=true

DB_CONNECTION=mysql
DB_HOST=db
DB_PORT=3306
DB_DATABASE=servas_db
DB_USERNAME=root
DB_PASSWORD=password12345678

#error
Starting Migration...
   Illuminate\Database\QueryException
  SQLSTATE[HY000] [2002] Connection refused (Connection: mysql, SQL: select
table_name as `name`, (data_length + index_length) as `size`, table_comment
as `comment`, engine as `engine`, table_collation as `collation` from
information_schema.tables where table_schema = 'servas_db' and table_type
in ('BASE TABLE', 'SYSTEM VERSIONED') order by table_name)
  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:829
    825▕                     $this->getName(), $query,
$this->prepareBindings($bindings), $e
    826▕                 );
    827▕             }
    828▕
  ➜ 829▕             throw new QueryException(
    830▕                 $this->getName(), $query,
$this->prepareBindings($bindings), $e
    831▕             );
    832▕         }
    833▕     }
      +39 vendor frames
  40  artisan:35
      Illuminate\Foundation\Console\Kernel::handle()
Creating caches...
   INFO  Configuration cached successfully.
   INFO  Blade templates cached successfully.
[08-Feb-2025 14:26:12] NOTICE: fpm is running, pid 19
[08-Feb-2025 14:26:12] NOTICE: ready to handle connections


On Fri, Feb 7, 2025 at 8:11 AM Henrik ***@***.***> wrote:

> Hey,
> this looks good, the database connection is now working and the migrations
> are complete.
> Now you have to generate an application key:
>
> docker exec -it servas php artisan key:generate --force
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/beromir/Servas/discussions/98#discussioncomment-12094185>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AARYC2QWKBHPGMV42JDTE7L2OSWGTAVCNFSM6AAAAABRKXU366VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTEMBZGQYTQNI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Regards,
Neil

**Comment by doc-huebi** [comment]
Hi,

I was sick for ablong time, sorry for not answering. I tried serves again with the following entries in portainer:

```
version: "3"

services:
  servas:
    image: beromir/servas
    container_name: Servas
    restart: unless-stopped
    ports:
      - 8070:80
    volumes:
      - /volume1/docker/servas/.env:/app/.env   
      - /volume1/docker/servas/database.sqlite:/app/database/database.sqlite
    environment:
      - APP_NAME=Servas
      - APP_ENV=production
      - APP_KEY=
      - APP_DEBUG=false
      - APP_URL=http://nuc:8070
      - SERVAS_ENABLE_REGISTRATION=true
      - SERVAS_SHOW_APP_VERSION=true      
      - DB_CONNECTION=sqlite
      - DB_DATABASE=/app/database/database.sqlite
      - DB_FOREIGN_KEYS=true  
```

The container started:

```
INF ts=1744789404.883354 msg=using config from file file=/etc/caddy/Caddyfile
INF ts=1744789404.8843818 msg=adapted config to JSON adapter=caddyfile
WRN ts=1744789404.8843944 msg=Caddyfile input is not formatted; run 'caddy fmt --overwrite' to fix inconsistencies adapter=caddyfile file=/etc/caddy/Caddyfile line=2
INF ts=1744789404.8853514 logger=admin msg=admin endpoint started address=localhost:2019 enforce_origin=false origins=["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]
INF ts=1744789404.8855333 logger=http.auto_https msg=automatic HTTPS is completely disabled for server server_name=srv0
INF ts=1744789404.8856068 logger=tls.cache.maintenance msg=started background certificate maintenance cache=0xc000594f00
WRN ts=1744789404.8858263 logger=http msg=HTTP/2 skipped because it requires TLS network=tcp addr=:80
WRN ts=1744789404.8858352 logger=http msg=HTTP/3 skipped because it requires TLS network=tcp addr=:80
INF ts=1744789404.8858395 logger=http.log msg=server running name=srv0 protocols=["h1","h2","h3"]
INF ts=1744789404.8900316 logger=tls msg=cleaning storage unit storage=FileStorage:/data/caddy
INF ts=1744789404.8912282 logger=tls msg=finished cleaning storage units
INF ts=1744789404.901623 logger=frankenphp msg=FrankenPHP started 🐘 php_version=8.3.19 num_threads=8 max_threads=8
INF ts=1744789404.901921 msg=autosaved config (load with --resume flag) file=/config/caddy/autosave.json
INF ts=1744789404.9019341 msg=serving initial configuration

```

Now generating the app key:

```
huebi@nuc:/volume1/docker/servas$ sudo docker exec -it Servas php artisan key:generate --force

   INFO  Application key set successfully.  

```


Looks fine so far, but http://nuc:8070/register is resulting in a 500er http code with no further entries in the log:

```
Oops! An Error Occurred

The server returned a "500 Internal Server Error".

Something is broken. Please let us know what you were doing when this error occurred. We will fix it as soon as possible. Sorry for any inconvenience caused.
```

At the start of the logs I can see:

```
   Illuminate\Database\QueryException 
  SQLSTATE[HY000] [14] unable to open database file (Connection: sqlite, SQL: PRAGMA foreign_keys = ON;)
  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:829
    825▕                     $this->getName(), $query, $this->prepareBindings($bindings), $e
    826▕                 );
    827▕             }
    828▕ 
  ➜ 829▕             throw new QueryException(
    830▕                 $this->getName(), $query, $this->prepareBindings($bindings), $e
    831▕             );
    832▕         }
    833▕     }
      +40 vendor frames 
  41  artisan:35
      Illuminate\Foundation\Console\Kernel::handle()
Creating caches...
   INFO  Configuration cached successfully.  
   INFO  Blade templates cached successfully.  
```

But the database is created successfully:

```
huebi@nuc:/volume1/docker/servas$ ls -al
total 20
drwxrwxrwx  4   82   82 4096 Apr 16 09:40 .
drwxr-xr-x 13 root root 4096 Nov  2 20:55 ..
-rw-rw-r--  1   82   82  458 Apr 16 10:18 .env
drwxrw-rw-  2 root root 4096 Apr 16 10:17 database.sqlite
drwxr-xr-x  2 root root 4096 Apr 16 09:36 env
huebi@nuc:/volume1/docker/servas$ 
```




I have no idea what I made wrong. Any help appreciated

//huebi

**Reply by beromir** [reply]
Hi,
sorry for the late reply.
Can you check the `.env` file inside the container and if the database file is there with correct permissions?
You can enter the container with:
```
docker exec -it Servas /bin/sh
```
